src/net/sourceforge/kolmafia/request/CurseRequest.java

```diff
@@ -64,59 +66,92 @@ public static final void parseResponse(final String location, final String respo
-    if (!m.find()) {
+    Matcher playerMatcher = CurseRequest.PLAYER_PATTERN.matcher(location);
+    if (!playerMatcher.find()) {
       return;
     }
-    AdventureResult item = ItemPool.get(StringUtilities.parseInt(m.group(1)), 1);
-    if (item.equals(CurseRequest.SMORE_GUN)) {
-      // When you "throw" a s'more gun at someone, marshmallows get used up
-      item = CurseRequest.MARSHMALLLOW;
+    String player = playerMatcher.group(1);
+
+    if (responseText.contains("That player could not be found")) {
+      KoLmafia.updateDisplay(
+          MafiaState.ERROR,
+          player + " evaded your thrown item by the unusual strategy of being nonexistent.");
+      return;
     }
 
-    m = CurseRequest.QTY_PATTERN.matcher(responseText);
-    if (!m.find()) {
+    if (responseText.contains("try again later")
+        || responseText.contains("cannot be used")
+        || responseText.contains("can't use this item")) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "Can't use the item on that player at the moment.");
       return;
     }
-    int qty = m.group(1) == null ? 0 : StringUtilities.parseInt(m.group(1));
-    qty = item.getCount(KoLConstants.inventory) - qty;
-    if (qty != 0) {
-      item = item.getInstance(qty);
-      ResultProcessor.processResult(item.getNegation());
+
+    if (responseText.contains("You can't fire")
+        || responseText.contains("That player has already been hit")) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "You cannot arrow that person.");
+      return;
     }
 
-    m = CurseRequest.PLAYER_PATTERN.matcher(location);
-    if (!m.find()) {
+    if (responseText.contains("They already know that skill.")) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "They already know that skill.");
       return;
     }
 
-    if (responseText.contains("You don't have that item")) {
-      KoLmafia.updateDisplay(MafiaState.ERROR, "Have not, throw not.");
+    if (responseText.contains("You've already trained somebody today.")) {
+      KoLmafia.updateDisplay(MafiaState.ERROR, "You've already trained somebody today.");
+      Preferences.setBoolean("_crimboTraining", true);
       return;
     }
 
-    if (responseText.contains("No message?") || responseText.contains("no message")) {
-      KoLmafia.updateDisplay(MafiaState.ERROR, "That item requires a message.");
+    Matcher itemMatcher = CurseRequest.ITEM_PATTERN.matcher(location);
+    if (!itemMatcher.find()) {
       return;
     }
+    AdventureResult item = ItemPool.get(StringUtilities.parseInt(itemMatcher.group(1)), 1);
 
-    if (responseText.contains("That player could not be found")) {
-      KoLmafia.updateDisplay(
-          MafiaState.ERROR,
-          m.group(1) + " evaded your thrown item by the unusual strategy of being nonexistent.");
+    // Special items require special processing
+    switch (item.getItemId()) {
+      case ItemPool.SMORE_GUN -> {
+        // When you "throw" a s'more gun at someone, marshmallows get used up
```

Comment:
For any other reviewer this is an Item and not the AdventureRequest of the same name that was deleted.